### PR TITLE
Rename persistence files for consistency 

### PR DESCRIPTION
**Pull request by longquanzheng**
**What changed?**
Manager is the first level, Store is the 2nd level.

**Why?**
For consistency and clearness 

**How did you test it?**
Existing test

**Potential risks**
No, only file name changes and struct name changes. 

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/reset/resetter.go [hunk 1]
```diff
@@ -23,7 +23,7 @@
 package reset
 
 import (
-	context "context"
+	"context"
 	ctx "context"
```
Comment (yycptt) [inline service/history/reset/resetter.go:27]: use only one `context` import?

File: common/persistence/utils/clusterMetadata.go [hunk 2]
```diff
@@ -18,7 +18,9 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package persistence
+package utils
```
Comment (yycptt) [inline common/persistence/utils/clusterMetadata.go:21]: nit: I kind of think having persistence in the package and import name is more informative than just `utils`, which is not a recommended package name (https://blog.golang.org/package-names#TOC_5.)

**Review by yycptt — COMMENTED** [review]
- Shall we also rename contents in sqlQueue.go also to `sqlQueueStore` to be consistence and match the filename?
- Is it possible to merge `dataVisibilityManagerInterfaces.go` into `dataManagerInterfaces.go`?

File: common/persistence/utils/clusterMetadata.go (hunk 2 above)
Comment (longquanzheng) [reply common/persistence/utils/clusterMetadata.go]: yeah good point. I dislike that too. I moved the ClusterMetadata to `cluster` package, and renamed this to `persistence-util`

File: service/history/reset/resetter.go (hunk 1 above)
Comment (longquanzheng) [reply service/history/reset/resetter.go]: Yeah good catch. Fixed

File: common/cluster/metadata.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ import (
 
 type (
 	// Metadata provides information about clusters
+	// TODO: use a more informative name, e.g. FailoverMonitor
```
Comment (yycptt) [inline common/cluster/metadata.go:36]: Does the TODO apply here? cluster.Metadata name seems reasonable to me for those methods.

Comment (longquanzheng) [reply common/cluster/metadata.go:36]: Yeah naming is controversial.  I just dislike the overloaded name of `metadata` everywhere. 
Maybe `ClusterMeta` will make more sense for me. 
I will remove this TODO and see if more people agree with me in the future.

File: common/persistence/persistenceutils/historyManagerUtil.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```
[the remark this answers is not in the document]
Comment (longquanzheng) [reply common/persistence/persistenceutils/historyManagerUtil.go]: Removed. Thanks

**Review by yycptt — APPROVED** [review]

**Pull request merged**